packages/rocketchat-geoip-plugin/server/server.js

```diff
@@ -0,0 +1,37 @@
+Accounts.onLogin(function(user) {
+	if (!user.user._id || !user.user.username) {
+		return;
+	}
+	const apiUrl = 'http://ip-api.com/json';
+	const response = HTTP.get(apiUrl).data;
+	if (!response || !response.country) {
+		return;
+	}
+	let room_id;
+	const is_channel_blacklisted = Meteor.users.findOne({
+		$and: [{ _id: user.user._id}, {'automatic_channels':{$elemMatch:{'name': response.country, 'blacklisted': true}}}]
```

Comment:
You should probably add to the array a field `type` to each subdocument, with name of the plugin for the automatic channel.